workers/data_refinery_workers/processors/management/commands/remove_invalid_qn_targets.py

```diff
@@ -0,0 +1,46 @@
+
+from django.core.management.base import BaseCommand
+from data_refinery_common.models import (
+    Sample,
+    ComputationalResult,
+    ComputationalResultAnnotation,
+    Organism
+)
+from data_refinery_common.logging import get_and_configure_logger
+from .create_qn_target import organism_can_have_qn_target
+logger = get_and_configure_logger(__name__)
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('--dry-run',
+                            help='Prints resulting actions without actually running them.',
+                            action='store_true')
+        parser.add_argument("--min",
+                            type=int,
+                            default=100,
+                            help=("Minimum number of processed microarray samples for each organism"))
+
+    def handle(self, *args, **options):
+        """ Takes care of removing all qn targets that have less than `min`
+        processed microarray samples. """
+        remove_invalid_qn_targets(options['min'], options['dry_run'])
+
+def remove_invalid_qn_targets(min_samples, dry_run):
+    computational_result_annotations = ComputationalResultAnnotation.objects\
+        .filter(data__is_qn=True)\
+        .order_by('-created_at')
+    qn_target_ids = []
+    for annotation in computational_result_annotations:
+        organism = Organism.objects.get(id=annotation.data.organism_id)
+        if not organism_can_have_qn_target(organism, min_samples):
+            # remove the referenced QN Target because it shouldn't have a qn target
+            logger.debug('Removing QN Target because it does not have enough samples on its biggest microarray platform',
+                         computational_result=annotation.result,
+                         organism=organism)
+            qn_target_ids.append(annotation.result.id)
+
+    if not dry_run:
+        # delete all invalid qn targets
+        ComputationalResult.objects.filter(id__in=qn_target_ids).delete()
```

Comment:
So I'm assuming that django will do a cascading delete here that will remove the ComputedFile object. However will that also delete the file from S3? I'm almost certain we don't have anything that will make it. Maybe we can use the pre_delete signal to have django handle this smoothly for us? https://docs.djangoproject.com/en/dev/ref/signals/#pre-delete

Reply:
Either that or just clean up the file manually. Whatever works for you :)